core/node/state_keeper/src/io/mempool.rs

```diff
@@ -265,7 +268,11 @@ impl StateKeeperIO for MempoolIO {
         Ok(())
     }
 
-    async fn reject(&mut self, rejected: &Transaction, error: &str) -> anyhow::Result<()> {
+    async fn reject(
+        &mut self,
+        rejected: &Transaction,
+        error: UnexecutableReason,
```

Comment:
Nit: Renaming the arg in the trait definition sort of implies that it should be renamed in implementations as well 🙃